Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs

```diff
@@ -282,15 +282,16 @@ public async Task<ActionResult> UploadPicture(IFormFile file, string userId)
         public async Task<IActionResult> GetNewsfeedAsync()
         {
             var email = User.FindFirst(ct => ct.Type.Contains("nameidentifier")).Value;
-            var followedUserEmails = _followsRepository.GetFollowedEmails(email);
-            followedUserEmails.Add(email);
-            // TODO: This query MUST be tested as EF may may not be able to convert it to sql!
-            // In case it doesn't work the posts repo will use the sql in NewsfeedAsync.
-            var newsfeedPosts = await _postsRepository
-            .Where(p => followedUserEmails.Contains(p.UserEmail))
-            .GroupBy(p => p.UserEmail)
-            .SelectMany(g => g.OrderByDescending(p => p.CreatedAt).Take(3))
-            .AsQueryable().ToListAsync();
+            var newsfeedPosts = await _postsRepository.NewsfeedAsync(email, 3);
+            // var followedUserEmails = _followsRepository.GetFollowedEmails(email);
+            // followedUserEmails.Add(email);
+            // // TODO: This query MUST be tested as EF may may not be able to convert it to sql!
+            // // In case it doesn't work the posts repo will use the sql in NewsfeedAsync.
+            // var newsfeedPosts = await _postsRepository
+            // .Where(p => followedUserEmails.Contains(p.UserEmail))
+            // .GroupBy(p => p.UserEmail)
+            // .SelectMany(g => g.OrderByDescending(p => p.CreatedAt).Take(3))
+            // .AsQueryable().ToListAsync();
```

Comment:
the implicit conversion from iqueryable to ienumerable is what destroys the "LINQ to SQL" connection.

Reply:
How would I fix this?

Reply:
the most straightforward ways with the current architecture are:
1. make IPostsRepository implement IQueryable instead of IEnumerable.
2. expose some IQueryable<Post> directly from the repository
3. encapsulate all logic that you want to translate into SQL in the repository class instead of trying to have it in the controller.

Reply:
the trouble with 1 or 2 is they'd be effectively just a pass-through to the DbSets though. you couldn't do it with the domain objects, since those aren't part of the EF model you can't do LINQ to SQL with them.

#3 involves moving the commented out code here into a repository method, and having it operate on the EF model.